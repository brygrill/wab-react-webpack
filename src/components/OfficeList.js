import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Segment, List } from 'semantic-ui-react';

import Loading from './Loading';

class Officelist extends Component {
  state = {
    loading: true,
    active: null,
    offices: [],
  };

  componentDidMount() {
    this.initOfficeLocationsListener();
    // query office feature service
  }

  initOfficesListener = () => {
    const { officesFeature } = this.props;
    officesFeature.on('click', evt => {
      const office = evt.graphic.attributes.office;
      this.setState({ active: office });
    });
  };

  render() {
    if (this.state.loading) {
      return <Loading />;
    }

    return (
      <Segment>
        <List divided animated selection>
          {this.state.offices.map((item) => {
            return (
              <List.Item key={item.id}>
                <List.Header>{item.office}</List.Header>
                {item.location}
              </List.Item>
            );
          })}
        </List>
      </Segment>
    );
  }
}

Officelist.propTypes = {
  officesFeature: PropTypes.object.isRequired,
};

export default Officelist;
