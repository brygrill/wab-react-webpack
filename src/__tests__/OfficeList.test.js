import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import OfficeList from '../components/OfficeList';

const offices = [
  { FID: 1, Office_Name: 'One', Address: '123 Main St' },
  { FID: 2, Office_Name: 'Two', Address: '123 Broad St' },
];

// mocking onclick handler here...
const officesFeature = {
  on() {
    console.log('on');
  },
};

describe('<OfficeList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <OfficeList officesFeature={officesFeature} offices={offices} />,
      div,
    );
  });

  it('shows list of offices', () => {
    const tree = renderer
      .create(<OfficeList officesFeature={officesFeature} offices={offices} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
