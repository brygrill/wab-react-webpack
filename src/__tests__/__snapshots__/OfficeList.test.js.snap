// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OfficeList /> shows list of offices 1`] = `
<div
  className="ui segment"
>
  <h3
    className="ui header"
  >
    EBA Offices
  </h3>
  <div
    className="ui animated divided selection list"
    role="list"
  >
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      <div
        className="header"
      >
        One
      </div>
      123 Main St
    </div>
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      <div
        className="header"
      >
        Two
      </div>
      123 Broad St
    </div>
  </div>
</div>
`;
